src/addagent/validate.c

```diff
@@ -550,9 +550,15 @@ double get_time_since_agent_disconnection(const char *id) {
         status = json_field->valuestring;
     }
 
-    json_field = cJSON_GetObjectItem(json_agent_info->child, "disconnected_time");
-    if (json_field && cJSON_IsNumber(json_field) && status && !strcmp(status, AGENT_CS_DISCONNECTED)) {
-        disconnected_time = json_field->valueint;
+    if (status) {
+        if (!strcmp(status, AGENT_CS_NEVER_CONNECTED)) {
+            return OS_INVALID;
```

Comment:
Memory leak: This return will leave **json_agent_info** allocated generating a memory leak.

Reply:
You're right